amy/workshops/models.py

```diff
@@ -2671,6 +2671,34 @@ def clean_event(
         elif not requirement.event_required and self.event:
             return self.get_not_required_error(requirement)
 
+        if self.event and hasattr(self, "trainee") and self.trainee:
```

Comment:
`trainee` is a FK, we should always have this attribute. It seems non-nullable, so maybe it's enough to check `if self.event:`?